sim/pkg/simulator/through_stock.go

```diff
@@ -36,10 +35,22 @@ func (s *stock) Count() uint64 {
 	return uint64(len(s.stock))
 }
 
-func (s *stock) EntitiesInStock() []*Entity {
+func (s *stock) EntitiesInStock() map[Entity]bool {
 	return s.stock
 }
 
+func (s *stock) GetEntityByNumber(number int) Entity {
+	counter := 0
+	fmt.Println("new loop")
```

Comment:
Is this debugging code?

Reply:
Yeah, I was just about to do a commit to remove it. Forgot to remove it.

Reply:
Removed it. See commit ab8fea3c6d8c8f9ae7d04e1406121f9279bec19d.